packages/admin/test/e2e/cli.feature.ts

```diff
@@ -0,0 +1,515 @@
+      const templateId1 = createTemplate('template1.json').split('Template id: ')[1];
+      const templateId2 = createTemplate('template2.json').split('Template id: ')[1];
+
+      const out = execCommand(
+        'get-templates',
+        ['--providerUrl', PROVIDER_URL],
+        ['--airnodeRrp', airnodeRrp.address],
+        ['--templateIds', [templateId1, templateId2]]
+      );
+
+      expect(JSON.parse(out)).toEqual([
+        {
+          airnodeId: '0x15e7097beac1fd23c0d1e3f5a882a6f99ecbcf2e0c1011d1bd43707c6c0ec717',
+          endpointId: '0x2605589dfc93c8f9c35eecdfe1e666c2193df30a8b13e1e0dd72941f59f9064c',
+          parameters: expect.anything(), // parameters are encoded
+        },
+        {
+          airnodeId: '0x73963232357ce7cf8bf536c4a855670977d110fb27a90b9584f5f40e32d72efb',
+          endpointId: '0x564d41afbc9dce517aaffc7b67d4e5edb05cb3b5f114b6984e16b46607f5726e',
+          parameters: expect.anything(), // parameters are encoded
+        },
+      ]);
+    });
+  });
+
+  describe('withdrawal', () => {
+    let requesterIndex: string;
+    let airnodeMnemonic: string;
+    let airnodeId: string;
+    let designatedWallet: ethers.Wallet;
+    let destinationWallet: ethers.Wallet;
+    const destinationBalance = async () => (await destinationWallet.getBalance()).toString();
+
+    beforeEach(async () => {
+      // Prepare for derivation of designated wallet - see test for designated wallet derivation for details
+      requesterIndex = await admin.createRequester(airnodeRrp, alice.address);
+      const airnodeWallet = ethers.Wallet.createRandom().connect(provider);
+      airnodeMnemonic = airnodeWallet.mnemonic.phrase;
+      const masterWallet = ethers.Wallet.fromMnemonic(airnodeMnemonic, 'm').connect(provider);
+      // Fund the master wallet - which will be used to set the airnode parameters
+      await deployer.sendTransaction({
+        to: masterWallet.address,
+        value: ethers.utils.parseEther('1'),
+      });
+      airnodeRrp = airnodeRrp.connect(airnodeWallet);
+      airnodeId = await admin.setAirnodeParameters(airnodeRrp, bob.address, []);
+
+      // Derive and fund the designated wallet
+      designatedWallet = await deriveDesignatedWallet(airnodeWallet, requesterIndex);
+      await deployer.sendTransaction({
+        to: designatedWallet.address,
+        value: ethers.utils.parseEther('1'),
+      });
+
+      // Create destination address
+      destinationWallet = ethers.Wallet.createRandom().connect(provider);
+      expect(await destinationBalance()).toBe('0');
+    });
+
+    it('can create and fulfill withdrawal request', async () => {
+      const countWithdrawalRequests = () =>
+        execCommand(
+          'count-withdrawal-requests',
+          ['--providerUrl', PROVIDER_URL],
+          ['--airnodeRrp', airnodeRrp.address],
+          ['--requesterIndex', requesterIndex]
+        );
+
+      // TODO: shouldn't this number start from 0? It's initialized to 1 in RequesterStore.sol
```

Comment:
@bbenligiray can you explain please? 
I'd be nice to add a comment in the contract, because it doesn't make much sense as of now.

Also, should the endpoint return `value - 1` ? Or maybe it should be documented that the number is actually +1 the actual number of requests.

Reply:
The number is initialized at `1` to avoid the first withdrawal request costing more gas than all of the following. I think renaming `requesterIndexToNoWithdrawalRequests` as `requesterIndexToNextWithdrawalRequestId` is a better solution.

Reply:
> The number is initialized at 1 to avoid the first withdrawal request costing more gas than all of the following.
Why? I don't get it why it would cost more gas if it started at `0`